drivers/serial/uart_nrfx_uarte.c

```diff
@@ -966,6 +1048,17 @@ static int uarte_nrfx_rx_buf_rsp(const struct device *dev, uint8_t *buf,
 	if (async_rx->buf == NULL) {
 		err = -EACCES;
 	} else if (async_rx->next_buf == NULL) {
+#ifdef CONFIG_HAS_NORDIC_DMM
```

Comment:
as above - this could should compile fine for legacy targets as well - `dmm_buffer_in_prepare` is mocked for them: https://github.com/zephyrproject-rtos/zephyr/blob/98e0e65069cf62297c4d03e0ad5b543374f5b735/soc/nordic/common/dmm.h#L134